### PR TITLE
feat: Login/SignUp/Logout implemented.

### PR DESCRIPTION
**Pull request by Victorportus**
Implemented Login/Signup/Logout. You will not be able to access Together or Profile if you are not logged in. 
Once you logout a new guest user is created.

### CODE REVIEW

**Comment by zeckkimon** [comment]
Task linked: [CU-86b0yptdp Login/Signup](https://app.clickup.com/t/86b0yptdp)

File: src/app/(tabs)/profile.tsx [hunk 1]
```diff
@@ -5,14 +5,36 @@ import Center from "@src/components/ui/Center";
...
 const ProfileScreen = () => {
   const isAuth = useUser((data) => data.isAuth);
+  const router = useRouter();
 
   if (!isAuth) {
     return <Redirect href="/provider-signup" />;
   }
 
+  const logout = () => {
+    const user = {
```
Comment (whamester) [inline src/app/(tabs)/profile.tsx]: no need to pass the user, just make the user params optional in the setAuth function

File: src/app/auth.tsx [hunk 2]
```diff
@@ -26,8 +28,35 @@ type AuthParams = {
...
+    // Check if all necessary params are available
+    if (!params || !params._id || !params.token) {
+      return;
+    }
+
+    const user = {
+      id: params._id,
+      deviceIds: [params.device_id].filter(Boolean) as string[],
+      currentDeviceId: params.device_id || null,
+      name: params.name || "User",
```
Comment (whamester) [inline src/app/auth.tsx]: instead of "User" just leave an empty string

File: src/app/(tabs)/profile.tsx (hunk 1 above)
Comment (Victorportus) [reply src/app/(tabs)/profile.tsx]: Fixed

File: src/app/auth.tsx (hunk 2 above)
Comment (Victorportus) [reply src/app/auth.tsx]: updated

File: src/services/authApi.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      daily_streak_counter: 0,
+      token: params.token,
+    };
+
+    setGuest(false);
+    setAuth(true, user);
+  };
+
+  const logout = async () => {
+    console.log("data", useUser.getState());
```
Comment (ReinhardtBotha) [inline src/services/authApi.tsx]: remove this console log

**Review by ReinhardtBotha — APPROVED** [review]
Nice! ✅

File: src/services/authApi.tsx (hunk 3 above)
Comment (Victorportus) [reply src/services/authApi.tsx]: removed

File: src/services/api.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const api = axios.create({
 
 api.interceptors.request.use(
   (request) => {
-    const { token } = useUser.getState().token;
+    const token = useUser.getState().token;
```
Comment (zeckkimon) [inline src/services/api.js]: This should be 

```javascript
const token = useUser.getState().user.token;
``` 
I missed that when I created the interceptor.

**Review by zeckkimon — COMMENTED** [review]
Please update the .gitignore.

File: src/services/authApi.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      daily_streak_counter: 0,
+      token: params.token,
+    };
+
+    setGuest(false);
+    setAuth(true, user);
+  };
+
+  const logout = async () => {
+    console.log("data", useUser.getState().user.token);
```
Comment (zeckkimon) [inline src/services/authApi.tsx]: Please remove this line

**Review by zeckkimon — APPROVED** [review]
Thanks Victor!

**Review by whamester — APPROVED** [review]

**Pull request merged**